apps/bfd-pipeline/bfd-pipeline-app/src/main/java/gov/cms/bfd/pipeline/app/PipelineManager.java

```diff
@@ -502,11 +568,12 @@ public PipelineJobOutcome call() throws Exception {
             "PipeLineJobOutcome interrupt failed with the the following: " + e.getMessage(), e);
         throw new InterruptedException("Re-firing job interrupt.");
       } catch (Exception e) {
+        // This will print the job failure + exception stacktrace
         handleJobFailure(jobRecord.getId(), e);
-
-        // Wrap and re-throw the failure.
-        throw new Exception("Re-throwing job failure.", e);
+        // If we've blown up without explicit recovery possible, stop the pipeline
+        stop();
```

Comment:
This was causing one of the primary issues; throwing this exception was killing the pipeline's execution, but without shutting the pipeline down, so it would just hang (with its metrics reporters still going looking as if it was still working).

With this adjustment we shut the pipeline down explicitly in the event of an error, however the pipeline is stuck on waiting for a thread in this case while still prevents proper shutdown. (More investigation needed to discern the remaining thread that's causing the job executor to be stuck from here)